lib/post_revisor.rb

```diff
@@ -159,7 +159,16 @@ def revise!(editor, fields, opts = {})
       end
     end
 
-    return false unless should_revise?
+    if !should_revise?
+      # the draft sequence is advanced here to handle the edge case where a
+      # user opens the composer to edit a post and makes some changes (which
+      # saves a draft), but then un-does the changes and clicks save. In this
+      # case, should_revise? returns false because nothing has really changed
+      # in the post, but we want to get rid of the draft so we advance the
+      # sequence.
+      advance_draft_sequence if !opts[:skip_advance_draft_seq]
+      return false
+    end
```

Comment:
This change (line 162 -> 171) isn't part of the fix, but it's somewhat related so I decided to include it here. I've moved the `advance_draft_sequence` method call out of the `post_changed?` method because `advance_draft_sequence` can change some state in the database and I'd not expect a method whose name ends with `?` to change any state.